### PR TITLE
Support filter attribute in the package source of the asset store

### PR DESCRIPTION
**Pull request by tgorgol**
**Description**

Changes proposed in this pull request:

- new `filter` attribute in the `source` of the Asset CRD allowing filtering package contents and storing only files matched by provided regex pattern
- docs update

**Related issue(s)**

#2654

### CODE REVIEW

File: docs/asset-store/docs/06-01-asset.md [hunk 1]
```diff
@@ -83,7 +84,7 @@ This table lists all possible parameters of a given resource together with their
 | **metadata.namespace** |    **YES**   | Defines the Namespace in which the CR is available. |
 | **spec.source.mode** |    **YES**   | Specifies if the asset consists of one file or a set of compressed files in the ZIP or TAR formats. Use `single` for one file and `package` for a set of files. |
 | **spec.source.url** |    **YES**   | Specifies the location of the file. |
-| **bucketref.name** |    **YES**   | Specifies the name of the bucket for storing the asset. |
+| **spec.source.filter** |    **NO**   | Specifies the regex pattern used to select files from package that should be stored. |
```
Comment (kazydek) [inline docs/asset-store/docs/06-01-asset.md]:
```suggestion
| **spec.source.filter** |    **NO**   | Specifies the regex pattern used to select files to store from the package. |
```

File: components/assetstore-controller-manager/pkg/loader/package_loader.go [hunk 2]
```diff
@@ -180,3 +186,21 @@ func (l *loader) createFile(src io.Reader, dst string, mode int64) error {
...
 }
+
+func (l *loader) filterFiles(files []string, filter string) ([]string, error) {
+	if filter == "" {
+		return files, nil
+	}
+
+	filtered := []string{}
+	for _, value := range files {
+		matched, err := regexp.MatchString(filter, value)
```
Comment (michal-hudy) [inline components/assetstore-controller-manager/pkg/loader/package_loader.go]: Compile regex outside the for-loop.

**Pull request merged**